src/java.base/share/classes/java/nio/file/Files.java

```diff
@@ -1522,13 +1522,11 @@ private static FileTypeDetector createDefaultFileTypeDetector() {
 
         // loads all installed file type detectors
         private static List<FileTypeDetector> loadInstalledDetectors() {
-            List<FileTypeDetector> list = new ArrayList<>();
-            ServiceLoader<FileTypeDetector> loader = ServiceLoader
-                .load(FileTypeDetector.class, ClassLoader.getSystemClassLoader());
-            for (FileTypeDetector detector: loader) {
-                list.add(detector);
-            }
-            return list;
+            return ServiceLoader.load(FileTypeDetector.class,
+                                      ClassLoader.getSystemClassLoader())
+                .stream()
+                .map(p -> p.get())
```

Comment:
This looks fine, the alternative is to use a method ref, as in map(ServiceLoader.Provider::get) if you want.

Reply:
So changed in c6f997e.